doc/performance-investigations/0010-web-server-worker-class/summary.md

```diff
@@ -0,0 +1,62 @@
+# Web server worker class: Async vs Threads vs Threads + GRPC
+
+We have only ever run the application with Gunicorn in async mode (workers). We should try threads if it is more performant, reliable or consistent. 
+Using threads also opens up other options:
+ - the ability to use GRPC for Datastore which is potentially more performant than HTTP
+ - the ability to use other Google libraries such as Pub/Sub which current is only performant under threads
+
+## Benchmark profile
+
+| Option                 | Value                        |
+|------------------------|------------------------------|
+| Requests file          | census_household_gb_eng.json |
+| Run time               | 30m                          |
+| User wait time minimum | 1                            |
+| User wait time maximum | 2                            |
+| Clients                | 64                           |
+| Hatch rate             | 64                           |
+| Number of workers      | 7                            |
+| Number of threads      | 7                            |
+
+This test replicates the current settings for the daily test.
+
+The version of runner tested: https://github.com/ONSdigital/eq-questionnaire-runner/releases/tag/v3.53.0
+
+## Results
+
+The tests were run multiple times to ensure the performance was consistent; however, I have only included a single sample per configuration in the outputs.
+The results below are the averages between all tests. Note response times are in milliseconds.
+
+|                          | Request Count | Failure Count | Median Response Time | Average Response Time | Min Response Time | Max Response Time | Average Content Size | Requests/s |
```

Comment:
With my maths hat on, you shouldn't mix median and average response time, it should be median and mean in this scenario as median is a classification of average

Reply:
I agree but kept it as it is for clarity as this is what locust reports and what we have used in past reports.